### ISSUE TITLE
Rerun is not working properly

### ISSUE THREAD

**Issue by mkumatag** [kubernetes/test-infra#25600]
**What happened**:

Rerun is not working properly, tried on multiple jobs and see same error:

<img width="1264" alt="image" src="https://user-images.githubusercontent.com/12646029/157804299-dc2ca4b0-fb00-430f-93a3-53da335c5950.png">

**What you expected to happen**:

Rerun should be successful

**How to reproduce it (as minimally and precisely as possible)**:

**Please provide links to example occurrences, if any**:

**Anything else we need to know?**:

**Comment by mkumatag** [comment]
I see this team number is coming all the way from here https://github.com/kubernetes/test-infra/blob/b87f0491d802c84568aa4890a0e2ef961b949970/config/prow/config.yaml#L150 and team name says: test-infra-admins

**Comment by mkumatag** [comment]
and I see team also exist here: https://github.com/orgs/kubernetes/teams/test-infra-admins, not sure how this prow is trying to fetch the members!

cc @BenTheElder anyone can help here to understand what is going wrong!

cc @chaodaiG @chizhg

**Comment by alvaroaleman** [comment]
@smg247 could this be related to your changes in the github client?

**Comment by smg247** [comment]
It is possible. Rerun (the way it is configured) is still be using the old `TeamHasMember` method though:
https://github.com/kubernetes/test-infra/blob/master/prow/apis/prowjobs/v1/types.go#L251-L259

and that uses `ListTeamMembers` which was working for `peribolos` up until I changed `peribolos` to no longer use that, so I believe that method is functioning properly.

**Comment by smg247** [comment]
@mkumatag is the 2009231 # test-infra-admins team part of the `kubernetes-sigs` org?

**Comment by alvaroaleman** [comment]
@smg247 nope, it is part of the Kubernetes org :(  We can not really assume that the org from the Prowjob is the org of the team for a global team config. I guess we will have to introduce a team_org setting or allows specifying the team as `orgname/teamname`?

This is a really subtile one. The `org` arg for `ListTeamMembers` was introduced by me for usage in GH apps auth, but the Openshift Prow deployment doesn't use teams, so if it was using the incorrect org, there is a good chance we wouldn't have noticed :(

As GitHub deferred the removal of the old apis, can we revert the changes to ListTeamMembers from https://github.com/kubernetes/test-infra/commit/4e92340cbb88e90c86b9bec79ab4785e42dead9f until we have a solution for this?

cc @chaodaiG @cjwagner for thoughts

**Comment by smg247** [comment]
@alvaroaleman yes, I can take `ListTeamMembers` back to just calling `/teams/{team_id}/members` for now since the sunset was postponed. PR forthcoming.

Updating the config to: `orgname/teamname` sounds like a good long term solution to me.
